cloudflare/fetch/request.go

```diff
@@ -0,0 +1,25 @@
+package fetch
+
+import (
+	"context"
+	"io"
+	"net/http"
+)
+
+// Request represents an HTTP request and is part of the Fetch API.
+// Docs: https://developers.cloudflare.com/workers/runtime-apis/request/
+type Request struct {
+	*http.Request
```

Comment:
If the field does not need to be public, I would avoid using embedding.
```go
	req *http.Request
```

Reply:
When we need to add a request header, it would be nice if the functions were inherited from `http.Request`. The code will be like below

```go
r, err := fetch.NewRequest(ctx, http.MethodGet, "https://", nil)
if err != nil {
	panic(err)
}
r.Header.Set("X-Test", "123")
```

Reply:
I think that when embedding an external type into a custom defined type, the boundary of the type's role becomes ambiguous because it inherits the functionality of the external type as it is.
Personally, I prefer to only expose the necessary functionality of the type through methods that need to be made public (such as `Header()` ), but it is currently difficult to determine which ones should be made public and which ones should not.

Since this `workers` package itself is experimental and subject to breaking changes in the future, I thought it would be acceptable to add functionality by embedding it in the form of a struct field this time.